data-prepper-plugins/otel-trace-group-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OTelTraceGroupPrepper.java

```diff
@@ -110,20 +113,17 @@ public OTelTraceGroupPrepper(final PluginSetting pluginSetting) {
         return recordsOut;
     }
 
-    private Map<String, String> searchTraceGroupByTraceIds(final Collection<String> traceIds) {
-        final Map<String, String> traceIdToTraceGroup = new HashMap<>();
+    private Map<String, TraceGroupWrapper> searchTraceGroupByTraceIds(final Collection<String> traceIds) {
+        final Map<String, TraceGroupWrapper> traceIdToTraceGroup = new HashMap<>();
         final SearchRequest searchRequest = createSearchRequest(traceIds);
 
         try {
             final SearchResponse searchResponse = restHighLevelClient.search(searchRequest, RequestOptions.DEFAULT);
             final SearchHit[] searchHits = searchResponse.getHits().getHits();
             Arrays.asList(searchHits).forEach(searchHit -> {
-                final DocumentField traceIdDocField = searchHit.field(OTelTraceGroupPrepperConfig.TRACE_ID_FIELD);
-                final DocumentField traceGroupDocField = searchHit.field(OTelTraceGroupPrepperConfig.TRACE_GROUP_FIELD);
-                if (Stream.of(traceIdDocField, traceGroupDocField).allMatch(Objects::nonNull)) {
-                    final String traceId = searchHit.field(OTelTraceGroupPrepperConfig.TRACE_ID_FIELD).getValue();
-                    final String traceGroup = searchHit.field(OTelTraceGroupPrepperConfig.TRACE_GROUP_FIELD).getValue();
-                    traceIdToTraceGroup.put(traceId, traceGroup);
+                Map.Entry<String, TraceGroupWrapper> entry = fromSearchHitToMapEntry(searchHit);
+                if (Objects.nonNull(entry)) {
```

Comment:
Same comment about `!= null`. However in this case I suggest an optional a bit further down, so the statement would instead be `if (entry.isPresent()) {`